icefoganalytics/travel-authorization
# Conform Travel Create/Edit To Wireframe

Fixes https://github.com/icefoganalytics/travel-authorization/issues/9

# Context

I notice that the existing form does not match the wireframe as seen in [UpdatedTA_ExpenseScreens.pdf](https://github.com/icefoganalytics/travel-authorization/blob/main/Design/UpdatedTA_ExpenseScreens.pdf), page 2.

![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/629ca444-a29f-4128-9325-5c73853772a4)

This ticket is the task of conforming the UI to the wireframe.

## Stretch Goal

Upgrade url mapping so that any section of the page can be reached by a url. This is an accessibility standard, and it also makes manual QA much faster.

# Screenshots

## New
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/98000d7e-56ab-479a-b7b9-1c382722968c)

## Old
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/5f2286da-3f7d-4f4c-9925-8f1bdc050f12)
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/20371266-ffc5-415a-9eb7-9c712d714436)
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/b7e09b83-889f-4ad2-bb6a-f94789c5c15b)

# Testing Instructions

1. Boot the db service via `dev up`
2. Boot the api service via `cd src/api && npm run start`
3. Boot the web service via `cd src/web && npm run start`
4. Log in to the app at http://localhost:8080.
5. Run the migrations at http://localhost:3000/migrate/latest
6. Go to http://localhost:8080/my-travel-requests and click on the "+ Travel Authorization" button.
7. Check that the action and tab matches the url i.e. http://localhost:8080/my-travel-requests/create/details
8. Check that breadcrumbs exist and work correctly. Note that the "create" breadcrumb redirects to create -> details so you can't actually go to http://localhost:8080/my-travel-requests/create
9. Check that the "Purpose" section corresponds to the wireframe.
10. Check that the "Details" section corresponds to the wireframe.
11. Check that the "Approvals" section is similar to the wireframe. Specific changes include remove of action and submit button. As this is a creation form, it doesn't make sense to have edit/review type buttons available. Separating component designs by action type and viewer type make things much easier to build and test.
12. Check that you can "Submit to Supervisor" and be redirected to the /my-travel-requests table page.
13. Repeat all the steps, and click "Save Draft" instead, check that you are again redirected to the /my-travel-requests table page.

"End Date"
1. Check that the "End Date" selector in the top header thing, is restricted to only permit values greater than or equal to the "Start Date".

"In Territory" flag
1. Check that the "In Territory" flag restricts the destinations list to only include "YT" locations. There are 6 selectors this effects.

"Trip Type"
1. Check that the "Trip Type" selector changes the number of stops you must enter data for. The final destination will remain the same as you swap trip types, but the inner values may be removed.

"Type of Accommodation"
1. Check that the "Type of Accommodation" selector in the "Details" panel has an "Other:" option.
2. The other option will add a text field for manual entry.

"Travel Method"
1. Check that all "Travel Method" fields in the "Details" panel support an "Other:" option. There are 6 fields, as section generated by selecting the "Trip Type" refers to a different set of components.
2. The other option will add a text field for manual entry.

"Expected Date return to work"
1. Check that the "Expected Date return to work" field of the "Details" panel is restricted to be greater than or equal to the "final destination" end date. 

"Pre-approved Travel Request?"
1. Check that the "Pre-approved Travel Request?" field in the "Details" panel is restricted to only showing content from the current user's department.

"Submit to"
1. Check that the "Submit to" field in the "Details" panel supports an email search against the YG government directory.
